src/bit.rs

```diff
@@ -19,7 +19,7 @@ impl BitIterator {
         let mask = bitmask(n);
         let ones = self.mask.trailing_ones();
         let mask_position = ones;
-        self.mask ^= mask << mask_position;
+        self.mask ^= mask.checked_shl(mask_position).unwrap_or(0);
```

Comment:
I suppose this is protecting the case were `mask` is already all ones? It would be nice to have a test for this